hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -665,18 +665,10 @@ private SerializerAdapter lookupGlobalSerializer(Class type) {
 
     private SerializerAdapter lookupJavaSerializer(Class type) {
         if (Externalizable.class.isAssignableFrom(type)) {
-            if (safeRegister(type, javaExternalizableAdapter) && !Throwable.class.isAssignableFrom(type)) {
-                logger.info("Performance Hint: Serialization service will use java.io.Externalizable for: " + type.getName()
-                        + ". Please consider using a faster serialization option such as DataSerializable.");
-            }
             return javaExternalizableAdapter;
         }
 
         if (Serializable.class.isAssignableFrom(type)) {
-            if (safeRegister(type, javaSerializerAdapter) && !Throwable.class.isAssignableFrom(type)) {
```

Comment:
I don't think that this is a good approach. This may hit performance. 

We should not register classes from temporary classloaders, or have a possibility to remove those values from registry.